### ISSUE TITLE
Add search by ZIPCODE

### ISSUE THREAD

**Issue by InciteDemocracy** [ResistanceCalendar/resistance-calendar-api#53]
Support proximity search by ZIP, City & State

Examples:

Events 50 miles from 10001
Events 100 miles from Austin, TX
Events in Montana 

![screenshot 2017-04-12 19 43 39](https://cloud.githubusercontent.com/assets/25624431/24991683/6716f134-1fb8-11e7-8ef3-d237cddfb7f6.png)

Docs:

While ODATA support seems to be [in the works](http://www.odata.org/blog/geospatial-data-support-in-odata/), it is not supported in the [odata-v4-mongodb](https://github.com/jaystack/odata-v4-mongodb) library we use, although it seems like it maybe somewhat supported in the underlying [parser](https://github.com/jaystack/odata-v4-parser).

One option is to possibly support geo.location in the odata-v4-mongodb library and convert it to the geojson search format:

```
{
   <location field>: {
     $near: {
       $geometry: {
          type: "Point" ,
          coordinates: [ <longitude> , <latitude> ]
       },
       $maxDistance: <distance in meters>,
       $minDistance: <distance in meters>
     }
   }
}
```
Another option may be to special case all this code in order to just get it working. It's also not clear how zipcodes are handled in ODATA, so may beed to be special cased anyway.

Some additional docs:
http://cannoneyed.github.io/geojson/
https://www.npmjs.com/package/zipcodes

**Comment by sdotson** [comment]
Looks like this issue might be closed from pr #61 that was merged in a few days ago.

**Comment by aaghevli** [comment]
City and zipcode search done. Thanks @KamillaKhabibrakhmanova !
